### PR TITLE
[release/6.0] Test on CentOS 7, not RedHat 7

### PR DESCRIPTION
**Pull request by dougbu**
# [release/6.0] Test on CentOS 7, not RedHat 7

- [x] You've read the [Contributor Guide](https://github.com/dotnet/aspnetcore/blob/main/CONTRIBUTING.md) and [Code of Conduct](https://github.com/dotnet/aspnetcore/blob/main/CODE-OF-CONDUCT.md).
- [x] You've included unit or integration tests for your change, where applicable.
- [x] You've included inline docs for your change, where applicable.
- [x] There's an open issue for the PR that you are making. If you'd like to propose a new feature or change, please open an issue to discuss the change or find an existing issue.

- https://github.com/dotnet/dnceng/issues/4298 part II/II
  - contributes to https://github.com/dotnet/dnceng/issues/3189
- use Docker container for CentOS, mimicking https://github.com/dotnet/runtime/pull/105063

## Description

Fixes part of dotnet/dnceng#4298

### CODE REVIEW

**Comment by dougbu** [comment]
hi @halter73, @BrennanConroy, @wtgodbe, and other @dotnet/aspnet-build members. I'm just visiting :grin:

oh @Tratcher is still in the CODEOWNERS file. nice and hi to you too

**Review by wtgodbe — APPROVED** [review]
LGTM, but are we planning on shipping another 6.0 release?

**Comment by dougbu** [comment]
on hold until release/6.0 branch opens again

**Comment by dougbu** [comment]
thanx @wtgodbe❗

**Pull request merged**
